nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/GetS3ObjectMetadata.java

```diff
@@ -0,0 +1,208 @@
+            .description("An object was found in the bucket at the supplied key")
+            .build();
+
+    public static Relationship REL_NOT_FOUND = new Relationship.Builder()
+            .name("not found")
+            .description("No object was found in the bucket the supplied key")
+            .build();
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return new HashSet<>(List.of(REL_FOUND, REL_NOT_FOUND, REL_FAILURE));
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final AmazonS3Client s3;
+        try {
+            s3 = getS3Client(context, flowFile.getAttributes());
+        } catch (Exception e) {
+            getLogger().error("Failed to initialize S3 client", e);
+            flowFile = session.penalize(flowFile);
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final String bucket = context.getProperty(BUCKET_WITH_DEFAULT_VALUE).evaluateAttributeExpressions(flowFile).getValue();
+        final String key = context.getProperty(KEY).evaluateAttributeExpressions(flowFile).getValue();
+        final Pattern attributePattern;
+        if (context.getProperty(ATTRIBUTE_INCLUDE_PATTERN).evaluateAttributeExpressions(flowFile).isSet()) {
+             attributePattern = Pattern.compile(context.getProperty(ATTRIBUTE_INCLUDE_PATTERN)
+                    .evaluateAttributeExpressions(flowFile).getValue());
+        } else {
+            attributePattern = null;
+        }
+
+        try {
+            Relationship relationship;
+
+            try {
+                ObjectMetadata metadata = s3.getObjectMetadata(bucket, key);
+                Map<String, Object> combinedMetadata = new HashMap<>(metadata.getRawMetadata());
+                combinedMetadata.putAll(metadata.getUserMetadata());
+
+                if (context.getProperty(METADATA_TARGET).getValue().equals(TARGET_ATTRIBUTES.getValue())) {
+                    String attributePrefix = context.getProperty(METADATA_ATTRIBUTE_PREFIX).getValue();
+                    Map<String, String> newAttributes = combinedMetadata
+                            .entrySet().stream()
+                            .filter(e -> {
+                                if (attributePattern == null) {
+                                    return true;
+                                } else {
+                                    return attributePattern.matcher(e.getKey())
+                                            .find();
+                                }
+                            })
+                            .collect(Collectors.toMap(e -> attributePrefix + e.getKey(), e -> e.getValue().toString()));
```

Comment:
The `toString()` approach works for most cases, but it is not optimal for `Date` objects due to the behavior of `java.util.Date.toString()`. Recommend checking for `Date` instances and using the time in seconds.
```suggestion
                            .collect(Collectors.toMap(e -> attributePrefix + e.getKey(), e -> {
                                final Object value = e.getValue();
                                final String attributeValue;
                                if (value instanceof Date dateValue) {
                                    attributeValue = Long.toString(dateValue.getTime());
                                } else {
                                    attributeValue = value.toString();
                                }
                                return attributeValue;
                            }));
```